sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py

```diff
@@ -106,6 +107,31 @@ def __init__(self, source, type, **kwargs): # pylint: disable=redefined-builtin
             raise ValueError("data and data_base64 cannot be provided at the same time.\
                 Use data_base64 only if you are sending bytes, and use data otherwise.")
 
+    @classmethod
+    def from_dict(cls, event, **kwargs):
+        # type: (Dict, Any) -> CloudEvent
+        """
+        Returns the deserialized CloudEvent object when a dict is provided.
+
+        :param event: The dict representation of the event which needs to be deserialized.
+        :type event: dict
+
+        :rtype: CloudEvent
+        """
+        return cls(
+        id=event.pop("id", None),
+        source=event.pop("source", None),
+        type=event.pop("type", None),
+        specversion=event.pop("specversion", None),
+        data=event.pop("data", None) or _decode(event.pop("data_base64", None)),
```

Comment:
In my understanding, to use the `from_dict` for `CloudEvent`, users should be doing this:

```python
dict = {
    data_base64 = b'<base64 bytes>'
}

# or

dict = {
    data = '<data>'
}
```

which means internally we do:
```
data=event.pop("data", None)
data_base64=event.pop("data_base64", None),
```

is this by design?

Reply:
I think it is by design, and mirrors [this](https://github.com/Azure/azure-sdk-for-python/pull/16441#discussion_r569031063). Similar to sending `data` that is of bytes type (sent as `data_base64` in the outgoing request), when `from_dict` is used to deserialize, it sets `data_base64` back to `data`.

Reply:
I'm thinking of the following two creation case which makes me feel the inconsistency, from intuition I would expect two cases to return the "same" `CloudEvent` :
```python
event_created_from_class = CloudEvent(data_base64=b'bytes_data')
event_created_from_class.data is None
event_created_from_class.data_base64 is not None

event_created_from_dict = CloudEvent.from_dict({'data_base64':b'bytes_data'})
event_created_from_class.data is not None
event_created_from_dict.data_base64 is None
```

apart from that, I notice that `from_dict` could take both `data` and `data_base64`, and if both are set `data_base64` gets ignored:
```python
# I think this should be an invalid input?
event_created_from_dict = CloudEvent.from_dict({'data': 'somedata', 'data_base64':b'bytes_data'})
```

Reply:
(I could also be wrong, might be missing some context here, just thinking out loud)

Reply:
That two ways that `data_base64` can be deserialized is a little confusing since, no matter what, only `data` will end up being set. Maybe clarifying this in the docstring of `from_dict` would be useful?

Reply:
we could also do a cross-language check

Reply:
update: we decided not to expose data_base64
https://github.com/Azure/azure-sdk-for-python/pull/16661